### PR TITLE
Changes to Results and List

### PR DESCRIPTION
**Pull request by jpsim**
- Make ACLs explicit
- Add SortDescriptor struct
- Add sorted(sortDescriptors:)
- Return optionals from min/max

### CODE REVIEW

File: Realm/RealmSwift/SortDescriptor.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+import Realm
+
+/**
+A `SortDescriptor` stores a property name and a sort order for use with
+`sorted(sortDescriptors:)`. It is similar to `NSSortDescriptor`, but supports
+only the subset of functionality which can be efficiently run by Realm's query
+engine.
+*/
+public struct SortDescriptor {
```
Comment (segiddins) [inline Realm/RealmSwift/SortDescriptor.swift]: any reason not to make this `equatable`?

Comment (jpsim) [inline Realm/RealmSwift/SortDescriptor.swift]: So far, I've only made things `Equatable` that were already equatable in Objective-C. If we make things equatable on the Swift side, then we should mirror that on ObjC side, which is out of scope for this IMO.

**Comment by segiddins** [comment]
This should now be ready for review, @alazier & @tgoyne

File: RealmSwift/Results.swift [hunk 2]
```diff
@@ -164,7 +164,16 @@ public final class Results<T: Object>: Printable, SequenceType {
...
-        return Results<T>(rlmResults.sortedResultsUsingProperty(property, ascending: ascending))
+        return sorted([SortDescriptor(property: property, ascending: ascending)])
+    }
+
+    /**
+    Returns `Results` with elements sorted by the given sort descriptors.
+    :param: sortDescriptors `SortDescriptor`s to sort by.
+    :returns: `Results` with elements sorted by the given sort descriptors.
+    */
+    public func sorted<S: SequenceType where S.Generator.Element == SortDescriptor>(sortDescriptors: S) -> Results<T> {
```
Comment (alazier) [inline RealmSwift/Results.swift]: we need to test with multiple sort descriptors

Comment (segiddins) [inline RealmSwift/Results.swift]: done

**Comment by segiddins** [comment]
@alazier @tgoyne please review?

File: RealmSwift/Tests/ListTests.swift [hunk 3]
```diff
@@ -186,7 +186,7 @@ class ListTests: TestCase {
         XCTAssertEqual(Int(1), array.filter(pred2).count)
     }
 
-    func testSort() {
+    func testSortWithFormat() {
```
Comment (tgoyne) [inline RealmSwift/Tests/ListTests.swift]: There's no formatting going on here...

**Comment by tgoyne** [comment]
:+1:

File: RealmSwift/Tests/ResultsTests.swift [hunk 4]
```diff
@@ -168,20 +172,32 @@ class ResultsTests: TestCase {
...
     }
 
+    func testSortWithDescriptor() {
+        let results = getAggregateableResults()
+
+        var sorted = results.sorted([SortDescriptor(property: "intCol", ascending: true)])
+        XCTAssertEqual(1, sorted[0].intCol)
+        XCTAssertEqual(2, sorted[1].intCol)
+
+        sorted = results.sorted([SortDescriptor(property: "doubleCol", ascending: false), SortDescriptor(property: "intCol", ascending: true)])
```
Comment (alazier) [inline RealmSwift/Tests/ResultsTests.swift]: You need at least 3 objects to ensure both descriptors are being used properly.

File: RealmSwift/Tests/ListTests.swift (hunk 3 above)
Comment (segiddins) [inline RealmSwift/Tests/ListTests.swift]: `testSortWithPropertyName`?

**Comment by segiddins** [comment]
This should now be good, please confirm?

**Comment by alazier** [comment]
:+1:

**Pull request merged**